core/pkg/ingress/controller/controller.go

```diff
@@ -620,9 +620,8 @@ func (ic *GenericController) getBackendServers() ([]*ingress.Backend, []*ingress
 			}
 
 			if rule.HTTP == nil &&
-				len(ing.Spec.TLS) == 0 &&
```

Comment:
@bprashanth yes. This validation tries to avoid creating a server without a valid configuration. Please check the context https://github.com/aledbf/ingress/blob/4cc24bd2164c139611c5691237505563744cad7e/core/pkg/ingress/controller/controller.go#L622